### DISCUSSION TITLE
Помогите сформировать логику работы с задачами.

### DISCUSSION THREAD

**Discussion by konditerov** [totumonline/totum-ru-issues-and-discussions#214] [Как это сделать?] [answered]
Добрый день, начал изучать тотум, в планах сделать для организации crm для ведения заказов.

Подскажите, возможно я ошибаюсь в своей логике.
CRM для производства памятников, многие процессы по заявке происходят параллельно и хотелось делать отметки в карточке и выводить в виде тегов. Т.е. что бы на импровизированном дашборде можно было увидеть что осталось выполнить 3 из 6 задач и понимать этап. Этапы: подтвержденная заявка - макет - ретушь - резка - гравировка - сборка - установка.
Сейчас в таблице реализовано это путем добавления слов в ячейку в начале заявки и удаление выполненного из ячейки.

Думал пойти по такому пути:
Новая заявка - подтвержденная заявка - заявка наделяется несколькими статусами, которые уже показываются разным сотрудникам с их ролями для одновлеменной работы над заявкой.
Статусы наделяются как теги, и по исполнению их - теги удаляются. Теги исполняют роль подзадач.

**Comment by erpic-ru** [comment]
В производственных компаниях я делаю общий набор статусов, отображающих ход основного бизнес-процесса. Например:

1. Создана – менеджером формируется список продукции к производству и ее характеристики.

2. Резерв материалов – материалы для производства ставятся в резерв и их видит снабжение 
<img width="1033" alt="Снимок экрана 2022-09-03 в 17 54 06" src="https://user-images.githubusercontent.com/91674114/188276884-fcf81ea9-9c57-4073-8dd7-0cd538879ff8.png">


3. Производство – заявка появляется в списке у начальников цехов и они могут отмечать этапы производства (резка, гравировка etc). Каждый видит только свои этапы и выбирает исполнителя его исполнителя. В этот момент списываются материалы и начисляется сдельная зарплата сотруднику:
<img width="1172" alt="Снимок экрана 2022-09-03 в 17 56 14" src="https://user-images.githubusercontent.com/91674114/188276232-c07c6944-3f6f-4610-8417-196b3e6775c2.png">

На странице общего списка заявок руководитель и менеджеры видят на каком этапе находится каждый заказ:
<img width="824" alt="Снимок экрана 2022-09-03 в 17 51 56" src="https://user-images.githubusercontent.com/91674114/188276001-721e7237-e133-407f-9907-d3aa566b6db5.png">. Можно представить это карточками канбана, но со списком  удобней работать.

4. Склад – отмечены последние этапы во всех позициях. Команда для менеджера по продажам.

5. Отгружено – создан и проведен УПД.

У вас как бы самый обычный линейный процесс, поэтому можно сделать просто последовательную полуавтоматическую смену статусов в зависимости от условий. Можно дать в доступ сотруднику как определенные строки по любому условию (статус, точка продаж и т.д), так и определенные столбцы (например, скрыть суммы в заказе, имя заказчика или этапы других подразделений).

**Comment by ttm-alexeypolunin** [comment]
Ну кстати интересная идея с тегами — я в понедельник подготовлю пример.

**Comment by ttm-alexeypolunin** [comment, marked as answer]
### [Видео на YouTube](https://youtu.be/pZYoPys4Ang)

Файл для загрузки: [https://raw.githubusercontent.com/totumonline/ttm-samples/main/samples/totum_sample_15.json.gz.ttm](https://raw.githubusercontent.com/totumonline/ttm-samples/main/samples/totum_sample_15.json.gz.ttm)

Инструкция по испорту: [https://youtu.be/UA7j28Ht5u4](https://youtu.be/UA7j28Ht5u4)
